packages/promotion/src/models/application-method.ts

```diff
@@ -35,25 +35,25 @@ export default class ApplicationMethod {
   id!: string
 
   @Property({ columnType: "numeric", nullable: true, serializer: Number })
-  value?: number | null
+  value?: string | null
```

Comment:
We need to figure out how to handle numbers via mikroORM model. columnType `numeric` creates a text column and we end up serializing that in the end to a number. This is how we followed in other examples as well. 

I think we should change these to the expected datatype, in this case a float or an integer. wdyt? @pKorsholm @adrien2p

Reply:
I think keeping this as numeric is the best choice (also what's recommended by postgres for monetary amounts) since it's exact by default (numeric and decimal are the only two exact numeric types and they're the equivalent and within sql standard).

postgres intentionally keeps the `numeric` type as a string to avoid any issues with rounding etc, and it requires serialization to work with the values as numbers, which mikro-orm doesn't provide ootb. We shouldn't use `integer` as the limit is too low for money amounts to be handled, especially when you consider decimals across different currencies where the value of a currency unit is relatively low. We could consider `bigInt` since the upper limit is quite obscene for most usecases in ecom but it doesn't support floating point numbers. 

That being said, for a lot of values we can probably get away with using `int` 😄 

typeorm issue for numerics: https://github.com/typeorm/typeorm/issues/873
mikro-orm issue for numerics: https://github.com/mikro-orm/mikro-orm/issues/2295
postgres numeric datatypes docs: https://www.postgresql.org/docs/9.4/datatype-numeric.html

Reply:
I believe we should store in Postgres as numeric with a very large precision, or as a text and our serializer would be a big number class.
For now, having them as number and serialize with Number might be enough, but ideally we replace that to a big number library like https://github.com/MikeMcl/bignumber.js/ in a near future.

Reply:
Interesting, thanks for the context @pKorsholm! 🎉 

Knowing this now, should we then stop considering these as numbers and work with string directly?

Reply:
Should we address this in a follow-up PR where we have an example at hand? Think it will be an easier starting point for the discussion.

And then in this PR, keep the numeric type + the serializer.

Reply:
yes, not necessary for this PR, I just wanted to raise this here.